### PR TITLE
Switch gif to tenor

### PR DESCRIPTION
**Pull request by nehayward**
This is a stable API that was just acquired by Google. I've found the results to be much better.

### Features
- Switched from Giphy to Tenor
- Switched to an `--explicit` flag.
- Updated unit tests.

## TODO
- [x] Update terraform 

[Create Key Here](https://tenor.com/developer/keyregistration)

### CODE REVIEW

File: bot/gif_command.go [hunk 1]
```diff
@@ -45,22 +43,28 @@ func NewGIFCommand(endpoint, token string, w io.Writer) cli.Command {
 			}
 
 			query := url.Values{}
-			query.Set("api_key", token)
-			query.Set("lang", "en")
-			query.Set("rating", c.String("rating"))
+			query.Set("key", key)
 			query.Set("q", strings.Join(args, " "))
```
Comment (zpatrick) [inline bot/gif_command.go]: Looking at the Tenor api docs, seems like we should add `limit=1` to the query?

File: bot/gif_command_test.go [hunk 2]
```diff
@@ -14,15 +15,15 @@ import (
 func TestGIF(t *testing.T) {
 	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		assert.Equal(t, "GET", r.Method)
-		assert.Equal(t, "/v1/gifs/search", r.URL.Path)
+		assert.Equal(t, "/v1/search", r.URL.Path)
 
 		query := r.URL.Query()
-		assert.Equal(t, "token", query.Get("api_key"))
-		assert.Equal(t, "pg", query.Get("rating"))
+		fmt.Println(query)
```
[the remark this answers is not in the document]
Comment (nehayward) [reply bot/gif_command_test.go]: Thanks!

File: bot/gif_command_test.go [hunk 3]
```diff
@@ -39,8 +40,8 @@ func TestGIF(t *testing.T) {
 	defer server.Close()
 
 	w := bytes.NewBuffer(nil)
-	cmd := NewGIFCommand(server.URL, "token", w)
-	if err := runTestApp(cmd, "!gif --rating pg dogs playing poker"); err != nil {
+	cmd := NewGIFCommand(server.URL, "key", w)
+	if err := runTestApp(cmd, "!gif --explicit dogs playing poker"); err != nil {
```
[the remark this answers is not in the document]
Comment (nehayward) [reply bot/gif_command_test.go]: Okay sweet, I'll add that.

**Comment by nehayward** [comment]
@zpatrick I more wanted to discuss if it was necessary or not. I think the limit is pretty high

File: bot/gif_command.go [hunk 4]
```diff
@@ -3,39 +3,37 @@ package bot
...
 		Usage:     "display a gif",
 		ArgsUsage: "args...",
 		Flags: []cli.Flag{
-			cli.StringFlag{
-				Name:  "rating",
-				Value: "pg-13",
-				Usage: "the MPAA-style rating for the search, e.g. 'pg', 'pg-13', 'r'",
+			cli.BoolFlag{
+				Name:  "explicit",
+				Usage: "This will turn off safe search https://tenor.com/gifapi/documentation#safesearch",
```
Comment (zpatrick) [inline bot/gif_command.go]: [nit] let's try and keep flag descriptions short & sweet. Maybe something just like "enable explicit content"?

File: bot/gif_command_test.go [hunk 5]
```diff
@@ -12,46 +12,62 @@ import (
 )
 
 func TestGIF(t *testing.T) {
-	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		assert.Equal(t, "GET", r.Method)
-		assert.Equal(t, "/v1/gifs/search", r.URL.Path)
+	cases := []struct {
```
Comment (zpatrick) [inline bot/gif_command_test.go]: so, the only thing we need to vary between test cases is the explicit flag being set or not; in that case, we could have our `cases` just be `[]bool{ true, false }`; but a little bit more polished might be:
```
cases := map[string]bool {
  "explicit flag disabled: false,
  "explicit flag enabled": true,
}

for name, explicit := range cases {
    t.Run(name, func(t *testing.T){ 
    ...
}
```

Then,  each case would have everything else the same, e.g. "dogs playing poker", key, url, etc.

File: bot/gif_command_test.go [hunk 6]
```diff
@@ -12,46 +12,57 @@ import (
...
-			Gifs: []GiphyGif{
-				{URL: "some url"},
-			},
-		}
+				query := r.URL.Query()
+				assert.Equal(t, "key", query.Get("key"))
+				assert.Equal(t, "dogs playing poker", query.Get("q"))
+				if explicit {
+					assert.Equal(t, "strict", query.Get("safesearch"))
+				}
```
Comment (zpatrick) [inline bot/gif_command_test.go]: We should also test the inverse:
```
expectedSafesearch := "strict"
if explicit {
    expectedSafesearch = ""
}

assert.Equal(t, expectedSafesearch, query.Get("safesearch"))
```

File: bot/gif_command_test.go [hunk 7]
```diff
@@ -12,46 +12,57 @@ import (
...
-	cmd := NewGIFCommand(server.URL, "token", w)
-	if err := runTestApp(cmd, "!gif --rating pg dogs playing poker"); err != nil {
-		t.Fatal(err)
-	}
+			server := httptest.NewServer(handler)
+			defer server.Close()
+
+			w := bytes.NewBuffer(nil)
+			cmd := NewGIFCommand(server.URL, "key", w)
+			if err := runTestApp(cmd, "!gif dogs playing poker"); err != nil {
```
Comment (zpatrick) [inline bot/gif_command_test.go]: The input will need to change for each case:
```
input := "!gif dogs playing poker"
if explicit {
    input = "!gif --explicit dogs playing poker"
}
```

**Pull request merged**